src/main/java/com/funnyboyroks/drawlib/renderer/ShapeRenderer.java

```diff
@@ -48,6 +47,13 @@ public void setColor(@NotNull Color color) {
         this.color = color;
     }
 
+    /**
+     * @param step_size The new step between two particles. Lower values can crash low-end devices
+     */
+    public void setStepSize(@NotNull double step_size) {
+        this.step_size = step_size;
```

Comment:
A bounds check here for step_size <= 0 would be great.

Something like this right before the `this.step_size = step_size` line:
```java
if (step_size <= 0) throw new IllegalArgumentException("step_size must be greater than 0.");
```